src/main/java/chess/domain/userAccess/UserAccessService.java

```diff
@@ -0,0 +1,60 @@
+package chess.domain.userAccess;
+
+import chess.domain.userAccess.room.Room;
+import chess.domain.userAccess.room.RoomDao;
+import chess.domain.userAccess.user.User;
+import chess.domain.userAccess.user.UserDao;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+public class UserAccessService {
+
+    private final UserDao userDao;
+    private final RoomDao roomDao;
+
+    public UserAccessService(UserDao userDao, RoomDao roomDao) {
+        this.userDao = userDao;
+        this.roomDao = roomDao;
+    }
+
+    public User findUserById(String userId) {
+        Optional<User> user = userDao.findUserById(userId);
+        if (user.isPresent()) {
+            return user.get();
+        }
+        User newUser = new User(userId);
+        userDao.createUser(newUser);
+        return newUser;
+    }
+
+    public boolean havaSavedRoom(User user) {
+        List<Room> rooms = findRoomsByUser(user);
+        if (rooms.isEmpty()) {
+            return false;
+        }
+        return true;
+    }
```

Comment:
```suggestion
    public boolean hasSavedRoom(User user) {
        List<Room> rooms = findRoomsByUser(user);
        return !rooms.isEmpty();
    }
```

https://www.quora.com/Which-one-is-correct-Someone-has-or-have
has 가 올바르다고 하네요!